src/java.base/share/classes/java/time/format/Parsed.java

```diff
@@ -446,6 +468,36 @@ private void resolveTimeFields() {
             updateCheckConflict(MINUTE_OF_DAY, HOUR_OF_DAY, mod / 60);
             updateCheckConflict(MINUTE_OF_DAY, MINUTE_OF_HOUR, mod % 60);
         }
+        if (dayPeriod != null) {
+            if (fieldValues.containsKey(HOUR_OF_DAY)) {
+                long hod = fieldValues.remove(HOUR_OF_DAY);
+                if (resolverStyle != ResolverStyle.LENIENT) {
+                    HOUR_OF_DAY.checkValidValue(hod);
+                }
+                updateCheckDayPeriodConflict(HOUR_OF_DAY, hod);
+            } else if (fieldValues.containsKey(HOUR_OF_AMPM)) {
+                long hoap = fieldValues.remove(HOUR_OF_AMPM);
+                if (resolverStyle != ResolverStyle.LENIENT) {
+                    HOUR_OF_AMPM.checkValidValue(hoap);
+                }
+                if (dayPeriod.includes((hoap + 12) * 60)) {
+                    hoap += 12;
+                }
+                updateCheckDayPeriodConflict(HOUR_OF_DAY, hoap);
+            } else {
+                long midpoint = dayPeriod.mid();
+                fieldValues.put(HOUR_OF_DAY, midpoint / 60);
+                fieldValues.put(MINUTE_OF_HOUR, midpoint % 60);
+                // dayPeriod precedes AmPm. Override it if exists.
+                if (fieldValues.containsKey(AMPM_OF_DAY)) {
+                    long ap = fieldValues.remove(AMPM_OF_DAY);
+                    if (resolverStyle != ResolverStyle.LENIENT) {
+                        AMPM_OF_DAY.checkValidValue(ap);
+                    }
+                    updateCheckDayPeriodConflict(AMPM_OF_DAY, midpoint / 720);
+                }
+            }
+        }
```

Comment:
Looking at the existing logic, the `AMPM_OF_DAY` field is completely ignored if there is no `HOUR_OF_AMPM` field. Thus, there is no validation to check `AMPM_OF_DAY` against `HOUR_OF_DAY`. This seems odd. (AMPM_OF_DAY = 0 and HOUR_OF_DAY=18 does not look like it throws an exception, when it probably should).

On solution would be for `AMPM_OF_DAY` to be resolved to a day period at line 427, checking for conflicts with any parsed day period. (a small bug fix behavioural change)

Reply:
There are cases where a period crosses midnight, e.g., 23:00-04:00 so it cannot validate am/pm, so I decided to just override ampm with dayperiod without validating.

Reply:
Pulling on this a little more.

As the PR stands, it seems that if the user passes in text with just a day-period of "AM" they get a `LocalTime` of 06:00 but if they pass in `AMPM_OF_DAY` of "AM" the get no `LocalTime` in the result. Is that right? If so, I don't think this is sustainable.

Thats why I think `AMPM_OF_DAY` will have to be resolved to a dayPeriod of "am" or "pm". If dayPeriod is more precise than `AMPM_OF_DAY`, then dayPeriod can silently take precedence

Reply:
I implemented what you suggested here in the latest PR, but that would be a behavioral change which requires a CSR, as "AM" would be resolved to 06:00 which was not before. Do you think it would be acceptable? If so, I will reopen the CSR and describe the change. (In fact some TCK failed with this impl)

Reply:
I find the whole "half way between the start and end" behaviour of day periods odd anyway, but if it is to be supported then it should be consistent as you've implemented. 

Another option I should have thought of earlier would be to simply not support the "half way between the start and end" behaviour of LDML in either dayPeriod or AM/PM. But since LDML defines it, you've implemented it, and it isn't overly harmful I think its OK to leave it in.

Would it be possible for STRICT mode to not have the "half way between the start and end" behaviour?

Reply:
I've had a look tonight, but found two more problems!

The comments at the start of `resolveTimeLenient()` indicate that setting the midpoint in `resolveTimeFields()` is a bad idea - it needs to be done inside `resolveTimeLenient()`. (This ensures user-defined fields can resolve to ChronoFields IIRC). Thus, the day period resolving would have to be split between the two methods. How important is the midpoint logic? Could it just be dropped?

Secondly, we need to ensure that "24:00 midnight" (using HOUR_OF_DAY only) correctly parses to the end of day midnight, not the start of day or an exception. This is related to the problem above. 

I _think_ (but haven't confirmed everything yet) that the only thing that should be in `resolveTimeFields()` is the resolution of HOUR_OF_AMPM + dayPeriod to HOUR_OF_DAY (with `dayPeriod` then being set to `null`). Everything else should go in `resolveTimeLenient()` - the midpoint logic in the first if block (where time fields are defaulted), and the validation logic at the end of the method.

Reply:
Thanks for the insights. I believe I fixed both of the issues with the new commit.

Reply:
Added a test case with the latest commit: https://github.com/openjdk/jdk/commit/a960804ff4d3f7df18e51fe59dcdcaf04542e10a